ui/app/adapters/ldap/library.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import NamedPathAdapter from 'vault/adapters/named-path';
+import { encodePath } from 'vault/utils/path-encoding-helpers';
+
+export default class LdapLibraryAdapter extends NamedPathAdapter {
+  getURL(backend, name) {
+    const base = `${this.buildURL()}/${encodePath(backend)}/library`;
+    return name ? `${base}/${name}` : base;
+  }
+
+  urlForUpdateRecord(name, modelName, snapshot) {
+    return this.getURL(snapshot.attr('backend'), name);
+  }
+  urlForDeleteRecord(name, modelName, snapshot) {
+    return this.getURL(snapshot.attr('backend'), name);
+  }
+
+  query(store, type, query) {
+    const { backend } = query;
+    return this.ajax(this.getURL(backend), 'GET', { data: { list: true } }).then((resp) => {
+      return resp.data.keys.map((name) => ({ name, backend }));
+    });
+  }
+  queryRecord(store, type, query) {
+    const { backend, name } = query;
+    return this.ajax(this.getURL(backend, name), 'GET').then((resp) => ({ ...resp.data, backend, name }));
+  }
+
+  fetchCheckOutStatus(backend, name) {
```

Comment:
I noticed on the api for LDAP there is a check-out management endpoint. And this method is related to the library status check. You will know more than me obviously, but would naming this `fetchLibraryStatus` make more sense in this context?

Reply:
I derived the method name from this description on the endpoint in the docs:

> This endpoint provides the check-out status of service accounts in a library set.

I haven't tested the endpoints yet but from what I understand the check-out endpoint allows for service accounts to be checked out from a library set.